clouds/postgres/modules/sql/quadbin/QUADBIN_DISTANCE.sql

```diff
@@ -0,0 +1,32 @@
+----------------------------
+-- Copyright (C) 2023 CARTO
+----------------------------
+
+CREATE OR REPLACE FUNCTION @@PG_SCHEMA@@.QUADBIN_DISTANCE
+(origin BIGINT, destination BIGINT)
+RETURNS BIGINT
+AS
+$BODY$
+    SELECT 
+        CASE WHEN origin IS NULL OR destination IS NULL THEN
+            NULL
+        ELSE
+            (WITH __quadbin_coords AS (
+            SELECT
+                @@PG_SCHEMA@@.QUADBIN_TOZXY(origin) AS origin_coords,
+                @@PG_SCHEMA@@.QUADBIN_TOZXY(destination) AS destination_coords
+            )
+            SELECT
+                CASE WHEN (origin_coords->>'z')::INT != (destination_coords->>'z')::INT THEN
+                    NULL
+                ELSE
+                    GREATEST(
+                        ABS((destination_coords->>'x')::DOUBLE PRECISION - (origin_coords->>'x')::DOUBLE PRECISION),
```

Comment:
minor thing: I guess we could cast the coordinates to to BIGINT as well, right?

Reply:
Actually the DOUBLE PRECISION was intended here. As I took the approach used in `QUADBIN_CENTER` but BIGINT makes more sense, will add it.